patches/0004-Add-OpenSSL-crypto-backend.patch

```diff
@@ -550,6 +560,54 @@ index f8485dc3ca1c29..9c1d3d279c472f 100644
  
  package fipsonly
  
+diff --git a/src/crypto/tls/key_schedule.go b/src/crypto/tls/key_schedule.go
+index d7f082c9ee1e04..ee9d4650e3523b 100644
+--- a/src/crypto/tls/key_schedule.go
++++ b/src/crypto/tls/key_schedule.go
+@@ -7,9 +7,11 @@ package tls
+ import (
+ 	"crypto/ecdh"
+ 	"crypto/hmac"
++	boring "crypto/internal/backend"
+ 	"errors"
+ 	"fmt"
+ 	"hash"
++	"internal/goexperiment"
+ 	"io"
+ 
+ 	"golang.org/x/crypto/cryptobyte"
+@@ -59,7 +61,16 @@ func (c *cipherSuiteTLS13) expandLabel(secret []byte, label string, context []by
+ 		panic(fmt.Errorf("failed to construct HKDF label: %s", err))
+ 	}
+ 	out := make([]byte, length)
+-	n, err := hkdf.Expand(c.hash.New, secret, hkdfLabelBytes).Read(out)
++	var r io.Reader
```

Comment:
I see we have a naming convention of a single character for 'obvious' types (such as 'r' for Reader), can I assume 'n' is for number and 'v' is value or void?

Reply:
It is idiomatic to use short names, up to a point. You got `r` and `n` right. `v` is not that common, I was thinking of it as `value`, but I now see that it is confusing, it would make more sense to give it a meaningful name, such as `ikm` (input key material), which is what `hkdf.Expands` conceptually returns.